include/faabric/util/queue.h

```diff
@@ -53,18 +53,31 @@ class Queue
             }
         }
 
-        T value = mq.front();
-        if (pop) {
-            mq.pop();
-            emptyNotifier.notify_one();
-        }
+        T value = std::move(mq.front());
+        mq.pop();
+        emptyNotifier.notify_one();
 
         return value;
     }
 
-    T peek(long timeoutMs = 0) { return doDequeue(timeoutMs, false); }
+    T* peek(long timeoutMs = 0)
```

Comment:
To make peek work when  the values in the queue aren't smart pointers, we need to return a pointer here to avoid copying.

Reply:
Nice